ambry-replication/src/main/java/com/github/ambry/replication/ReplicaThread.java

```diff
@@ -238,6 +248,7 @@ void controlReplicationForPartitions(Collection<PartitionId> ids, boolean enable
     } finally {
       lock.unlock();
     }
+    terminateCurrentContinuousReplicationCycle = true;
```

Comment:
In case of exceptions do you want this boolean set?

Reply:
this boolean will tell when to stop a current running cycle. In case of exception too , we can have some partition added /removed from list of partition to replicate , so we should terminate cycle as cycle can be very long (hours)

Reply:
In that case should this be in finally block?

Reply:
updated, moved to finally.